libraries/serverHelpers/oauthRequestHandlers.js

```diff
@@ -0,0 +1,54 @@
+const fetch = require('node-fetch');
+const querystring = require('querystring');
+
+const oauthRefreshRequestHandler = (req, res) => {
+    const refreshUrl = req.params[0]; // TODO: get this from the tool somehow to prevent leaking secret to any supplied url
```

Comment:
If this is expected to always be an onshape url there could be an allowList of known urls to which we're okay with sending oauth requests. Any new developer of an oauth-enabled tool would then locally modify this allowList and eventually get us to ad the new domain to all edge servers

Reply:
I was thinking of having the allowList be part of the secrets file for the add-on, since this is designed to be used with anything that supports OAuth, not just Onshape.

Reply:
Thumbs up emoji